guidance/models/_engine/_engine.py

```diff
@@ -401,13 +401,13 @@ def get_top_k(_probs: NDArray, _k: int = 5) -> list[GenToken]:
         # compute top-k with masking
         masked_top_k: list[GenToken] = []
         if mask is not None:
-            # shift logits to [0 - max] range first and apply mask
-            masked_logits = (logits - np.min(logits)) * np.frombuffer(mask, dtype=np.uint8)
-            masked_probs = (
-                softmax(masked_logits)
-                if temperature < 0.0001
-                else softmax(masked_logits / temperature)
-            )
+            masked_logits = logits + np.frombuffer(mask, dtype=np.uint8)
+            if temperature < 0.0001:
```

Comment:
Why this particular value for temperature?

Reply:
0.0001 was there before, and I don't know why this particular value was chosen.  I assume it's there just to handle temp==0.0, which would create NaN values otherwise.

Reply:
Ahh yes. It appears in a few places... can we at least factor it out as a TEMPERATURE_EPSILON constant, or something like that?

Reply:
We can!  Done.